runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -320,125 +238,303 @@ private NodeStatus getLogUnitNodeStatusInLayout(Layout layout, String server) {
-     * @param allServers        All servers in the layout.
-     * @return Map of nodes mapped to their status.
+     *  In this sense a cluster can be STABLE with all nodes UP, while not being available for a
+     *  client, as all connections from the client to the cluster nodes are down, showing in this
+     *  case connectivity status to all nodes as UNRESPONSIVE.
+     *
+     *  The ClusterStatusReport consists of the following:
+     *
+     *  CLUSTER-SPECIFIC STATUS
+     *  - clusterStatus: the cluster status a perceived by the system's layout.
+     *  STABLE, DEGRADED, DB_SYNCING or UNAVAILABLE
+     *  - nodeStatusMap: each node's status as perceived by the system's layout.
+     *  (UP, DOWN or DB_SYNC)
+     *  - Cluster Status Reliability: STRONG_QUORUM, WEAK_NO_QUORUM or UNAVAILABLE
+     *
+     *  CLIENT-CLUSTER SPECIFIC STATUS:
+     *  - clientServerConnectivityStatusMap: the connectivity status of this client to the cluster.
+     *    (RESPONSIVE, UNRESPONSIVE).
+     *
+     * @return ClusterStatusReport
      */
-    private Map<String, NodeStatus> createNodeStatusMap(Layout layout,
-                                                        Set<String> responsiveServers,
-                                                        Set<String> allServers) {
-        Map<String, NodeStatus> nodeStatusMap = new HashMap<>();
-        for (String server : allServers) {
-            nodeStatusMap.put(server, NodeStatus.DOWN);
-            if (responsiveServers.contains(server)) {
-                nodeStatusMap.put(server, getLogUnitNodeStatusInLayout(layout, server));
+    public ClusterStatusReport getClusterStatus() {
+        Layout layout;
+        ClusterStatusReliability statusReliability = ClusterStatusReliability.STRONG_QUORUM;
+        List<String> layoutServers = runtime.getLayoutServers().stream()
+                .map(endpoint -> NodeLocator.getLegacyEndpoint(NodeLocator.parseString(endpoint)))
+                .collect(Collectors.toList());
+
+        try {
+            // Obtain Layout from layout servers:
+            //      Attempt to get layout from quorum (majority of nodes),
+            //            ---if we fail to get from quorum----
+            //      we are unable to provide a reliable state of the cluster from the layout,
+            //      therefore, the cluster status is UNAVAILABLE.
+
+            // The initial list of layoutServers is obtained from the runtime, this list might
+            // not be complete as it can be based on a a minimum of servers required for bootstrap,
+            // by setting the discoverLayoutServers flag, we request layouts from all available
+            // layout servers present in the actual layout file.
+            Map<String, Layout> layoutMap = getLayouts(layoutServers, true);
+
+            int serversLayoutResponses = layoutMap.size();
+            int quorum = runtime.getLayoutView().getQuorumNumber();
+
+            if (serversLayoutResponses == 0) {
```

Comment:
you could simplify it: layoutMap.isEmpty()

Reply:
Anyway we need it for the second if condition, < quorum.

Reply:
yes, just use that number for second condition